sklearn/base.py

```diff
@@ -590,6 +590,9 @@ class attribute, which is a dictionary `param_name: list of constraints`. See
         the docstring of `validate_parameter_constraints` for a description of the
         accepted constraints.
         """
+        if get_config()["skip_parameter_validation"]:
```

Comment:
Should the skip be here or in `validate_parameter_constraints`?

Reply:
Here: we want to skip as fast as possible. For the estimators we want to skip before calling `get_params` and for functions we want to skip before calling `signature`